File: experiments/issue-302/probe.mjs
// What the JavaScript parser says about documents that do not parse.
import { Parser } from '../../js/src/index.js';

const docs = ['# ok line\n# break: two\nci_gate x\n', 'a: b: c', 'a (b\n', 'a b)\n', ':'];
for (const doc of docs) {
  try {
    const links = new Parser().parse(doc);
    console.log(`${JSON.stringify(doc)} -> PARSED ${links.length} links`);
  } catch (error) {
    console.log(`${JSON.stringify(doc)} -> offset ${error.offset}`);
    console.log(error.message);
  }
}

File: experiments/issue-302/java-probe/Probe.java
import io.github.linkfoundation.linksnotation.Parser;
import java.util.List;

public class Probe {
  public static void main(String[] args) {
    String[] docs = {"# ok line\n# break: two\nci_gate x\n", "a: b: c", "a (b\n", "a b)\n", ":"};
    for (String doc : docs) {
      try {
        List<?> links = new Parser().parse(doc);
        System.out.println(quote(doc) + " -> PARSED " + links.size() + " links");
      } catch (Throwable e) {
        System.out.println(quote(doc) + " -> " + e.getClass().getSimpleName() + ": " + e.getMessage());
      }
    }
  }

  static String quote(String s) {
    return "\"" + s.replace("\n", "\\n") + "\"";
  }
}
